packages/client/src/features/tools/edge-creation/edge-creation-tool.ts

```diff
@@ -131,30 +143,61 @@ export class EdgeCreationToolMouseListener extends DragAwareMouseListener {
     override mouseOver(target: SModelElement, event: MouseEvent): Action[] {
         const newCurrentTarget = findParentByFeature(target, isConnectable);
         if (newCurrentTarget !== this.currentTarget) {
+            this.pendingDynamicCheck = false;
             this.currentTarget = newCurrentTarget;
             if (this.currentTarget) {
                 if (!this.isSourceSelected()) {
-                    this.allowedTarget = this.isAllowedSource(newCurrentTarget);
+                    this.allowedTarget = this.canConnect(newCurrentTarget, 'source');
                 } else if (!this.isTargetSelected()) {
-                    this.allowedTarget = this.isAllowedTarget(newCurrentTarget);
-                }
-                if (this.allowedTarget) {
-                    const action = !this.isSourceSelected()
-                        ? cursorFeedbackAction(CursorCSS.EDGE_CREATION_SOURCE)
-                        : cursorFeedbackAction(CursorCSS.EDGE_CREATION_TARGET);
-                    return [action];
+                    this.allowedTarget = this.canConnect(newCurrentTarget, 'target');
                 }
+            } else {
+                this.allowedTarget = false;
             }
-            return [cursorFeedbackAction(CursorCSS.OPERATION_NOT_ALLOWED)];
+            return [this.updateEdgeFeedback()];
         }
         return [];
     }
 
-    protected isAllowedSource(element: SModelElement | undefined): boolean {
-        return element !== undefined && isConnectable(element) && element.canConnect(this.proxyEdge, 'source');
+    protected updateEdgeFeedback(): Action {
+        if (this.allowedTarget) {
+            const action = !this.isSourceSelected()
+                ? cursorFeedbackAction(CursorCSS.EDGE_CREATION_SOURCE)
+                : cursorFeedbackAction(CursorCSS.EDGE_CREATION_TARGET);
+            return action;
+        }
+        return cursorFeedbackAction(CursorCSS.OPERATION_NOT_ALLOWED);
+    }
+
+    protected canConnect(element: SModelElement | undefined, role: 'source' | 'target'): boolean {
+        if (!element || !isConnectable(element) || !element.canConnect(this.proxyEdge, role)) {
+            return false;
+        }
+        if (!this.isDynamic(this.proxyEdge.type)) {
+            return true;
+        }
+        const sourceElement = this.source ?? element;
+        const targetElement = this.source ? element : undefined;
+
+        this.pendingDynamicCheck = true;
+        // Request server edge check
+        this.actionDispatcher
```

Comment:
I wonder whether we should provide some feedback on the cursor while we are waiting for the result (e.g., wait cursor) as we do not know exactly how long the calculation may take.